src/core/reference/include/openvino/reference/acosh.hpp

```diff
@@ -4,25 +4,35 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cmath>
-#include <cstddef>
 
-namespace ngraph {
-namespace runtime {
+#include "openvino/reference/utils/type_util.hpp"
+
+namespace ov {
 namespace reference {
-template <typename T, typename std::enable_if<!std::is_integral<T>::value, bool>::type = true>
-void acosh(const T* arg, T* out, size_t count) {
-    for (size_t i = 0; i < count; i++) {
-        out[i] = static_cast<T>(std::acosh(arg[i]));
-    }
+namespace func {
+template <class T, typename std::enable_if<ov::is_floating_point<T>()>::type* = nullptr>
+T acosh(const T in) {
```

Comment:
Why do we have different implementations for `acos` and `acosh`?
Should we align the approach?

Reply:
I didn't want change functionality to current implementation only refactor it and assumed that cast is intentional.
I will make separate PR for it as rounding gives more accurate result for integral values.